tools/rtpsrelay/PublicationListener.cpp

```diff
@@ -0,0 +1,131 @@
+#include "PublicationListener.h"
+
+#include "utility.h"
+
+PublicationListener::PublicationListener(OpenDDS::DCPS::DomainParticipantImpl* participant,
+                                         RtpsRelay::WriterEntryDataWriter_ptr writer,
+                                         const AssociationTable& association_table) :
+  participant_(participant),
+  writer_(writer),
+  association_table_(association_table)
+{}
+
+void PublicationListener::on_requested_deadline_missed(DDS::DataReader_ptr /*reader*/,
+                                                       const DDS::RequestedDeadlineMissedStatus & /*status*/)
+{}
```

Comment:
The empty implementations that belong to all (or most of the) four listener types could be moved to a base class.